pkg/network/node/networkpolicy.go

```diff
@@ -304,6 +339,49 @@ func (np *networkPolicyPlugin) SyncVNIDRules() {
 	}
 }
 
+// Match namespaces against a selector, using a cache so that, eg, when a new Namespace is
+// added, we only figure out if it matches "name: default" once, rather than recomputing
+// the set of namespaces that match that selector for every single "allow-from-default"
+// policy in the cluster.
+//
+// Yes, if a selector matches against multiple labels then the order they appear in
+// cacheKey here is non-deterministic, but that just means that, eg, we might compute the
+// results twice rather than just once, and twice is still better than 10,000 times.
+func (np *networkPolicyPlugin) selectNamespacesInternal(selector labels.Selector) map[string]uint32 {
+	cacheKey := selector.String()
+	match := np.nsMatchCache[cacheKey]
+	if match == nil {
+		match = &npCacheEntry{selector: selector, matches: make(map[string]uint32)}
+		for vnid, npns := range np.namespaces {
```

Comment:
This is a bit different from https://github.com/openshift/sdn/pull/36/files#diff-770c2f485d06d6a759ebe04d107047eaR317 - any reason for that?

Reply:
> This ended up being slightly different from the 4.3/4.2/4.1 version because 3.11 doesn't have namespaceSelector+podSelector support so the selectNamespacesInternal changes didn't apply completely cleanly and had to be rewritten a bit.

Reply:
OK, as long as it wasn't accidentally a backport of an old patch.